python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -1448,6 +1448,7 @@ def test_write_to_dataset_no_partitions_s3fs(
     "ignore:'ParquetDataset:FutureWarning",
     "ignore:'partition_filename_cb':FutureWarning")
 @pytest.mark.pandas
+@pytest.mark.dataset
```

Comment:
This should not be required because we are already marking all as requiring dataset: https://github.com/apache/arrow/blob/main/python/pyarrow/tests/test_dataset.py#L59

Reply:
That's a different file; this one just requires parquet.